### DISCUSSION TITLE
How to reproduce Crazyfile experiment

### DISCUSSION THREAD

**Discussion by K4IN1** [TinyMPC/TinyMPC#40] [Q&A] [answered]
I am conducting drone experiments using motion capture and crazyswarm 2. I want to know how to enable tinympc in the Crazyflie firmware and achieve dynamic obstacle avoidance or posture recovery. (As shown in the video)
I found a repo about the [firmware](https://github.com/RoboticExplorationLab/tinympc-crazyflie-firmware/tree/main),but I didn't flash it
I am not very familiar with firmware. It seems that tinympc is implemented in the App layer. In which directory should I compile it?
How should obstacles be configured in crazyswarm?
Any form of answer is acceptable, thank you

**Comment by khairesearch** [comment, marked as answer]
Hi,

Please take a look at https://github.com/A2R-Lab/tinympc-crazyflie/tree/main
